### PR TITLE
Conform URL to ExpressibleByStringLiteral

### PR DESCRIPTION
**Pull request by wmcginty**
#60 

Figured I'd throw a PR up so we can see if this is worth adding. I have one main concern which is the possibility of multiple modules conforming `URL` to `ExpressibleByStringLiteral` and almost believe this is something that should be done on a project by project level. 

Thoughts?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,10 @@
 [Tyler Milner](https://github.com/tylermilner)
 [#65](https://github.com/BottleRocketStudios/iOS-UtiliKit/pull/65)
 
+* Conform URL to ExpressibleByStringLiteral
```
Comment (tylermilner) [inline CHANGELOG.md]: We might just add a quick example here to make it easier for consumers to understand.
```
* Conform `URL` to `ExpressibleByStringLiteral` (e.g. `let myURL = URL("https://apple.com")`)
```

File: Sources/UtiliKit/General/URL+Extensions.swift [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+//
+//  Created by William McGinty on 7/22/19.
+//  Copyright © 2019 Bottle Rocket Studios. All rights reserved.
+//
+
+import Foundation
+
+extension URL: ExpressibleByStringLiteral {
+    
+    /// Creates a `URL` from a string literal object. This initializer will `fatalError` if given an invalid URL string.
```
Comment (tylermilner) [inline Sources/UtiliKit/General/URL+Extensions.swift]: Also might want to add some additional info on how this is intended to be used (only for static "pre-baked" strings and not dynamically generated ones).
```
/// Creates a `URL` from a string literal object. This initializer will `fatalError` if given an invalid URL string. For this reason, you should only use this for URLs that are baked-in to your app at compile time. We do *not* recommend using this to create URLs from strings downloaded at run time.
```

**Review by tylermilner — COMMENTED** [review]
It seems like every project creates a variant of this at some point so I think it's worth including here. If we find there are conflicts with other modules, then we can break this into its own subspec so that it can be brought in on a case-by-case basis.

File: CHANGELOG.md (hunk 1 above)
Comment (wmcginty) [reply CHANGELOG.md]: Done

File: Sources/UtiliKit/General/URL+Extensions.swift (hunk 2 above)
Comment (wmcginty) [reply Sources/UtiliKit/General/URL+Extensions.swift]: I believe the compiler will enforce that will it not? It has to be a string literal so anything dynamic won't pass the type check.

Comment (tylermilner) [reply Sources/UtiliKit/General/URL+Extensions.swift]: I'm not sure exactly when the compiler will enforce things. This seems to work okay, but I would argue that it's not how this extension should be used:
```swift
let page = 0
let myURL = URL("http://apple.com?page=\(page)")
```

Comment (wmcginty) [reply Sources/UtiliKit/General/URL+Extensions.swift]: Fair enough, will add a 'use for static pre-defined URLs only' comment.

Comment (tylermilner) [reply Sources/UtiliKit/General/URL+Extensions.swift]: Although maybe that's a poor example since `page` is known at compile-time...

**Review by tylermilner — DISMISSED** [review]

**Comment by wmcginty** [comment]
@tylermilner Looks like I needed to fix a conflict. Can you give this one last look over before merging?

**Review by tylermilner — APPROVED** [review]

**Pull request merged**